core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,106 @@
-        FETCH_ISSUES_URL, {
-          exploration_id: explorationId
-        });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughIssuesBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private urlInterpolationService: UrlInterpolationService,
+    private playthroughIssueObjectFactory: PlaythroughIssueObjectFactory) { }
 
-    var getFullPlaythroughUrl = function(expId, playthroughId) {
-      return UrlInterpolationService.interpolateUrl(
-        FETCH_PLAYTHROUGH_URL, {
-          exploration_id: expId,
-          playthrough_id: playthroughId
-        });
-    };
+  private cachedIssues = null;
 
-    var getFullResolveIssueUrl = function(explorationId) {
-      return UrlInterpolationService.interpolateUrl(
-        RESOLVE_ISSUE_URL, {
-          exploration_id: explorationId
-        });
-    };
-    return {
-      fetchIssues: function(explorationId, explorationVersion) {
-        if (cachedIssues !== null) {
-          return $q.resolve(cachedIssues);
-        } else {
-          return $http.get(getFullIssuesUrl(explorationId), {
-            params: {
-              exp_version: explorationVersion
-            }
-          }).then(function(response) {
-            var unresolvedIssueBackendDicts = response.data;
-            cachedIssues = unresolvedIssueBackendDicts.map(
-              PlaythroughIssueObjectFactory.createFromBackendDict);
-            return cachedIssues;
-          });
-        }
-      },
-      fetchPlaythrough: function(expId, playthroughId) {
-        return $http.get(getFullPlaythroughUrl(expId, playthroughId)).then(
-          function(response) {
-            var playthroughBackendDict = response.data;
-            return PlaythroughObjectFactory.createFromBackendDict(
-              playthroughBackendDict);
-          });
-      },
-      resolveIssue: function(issueToResolve, expId, expVersion) {
-        return $http.post(getFullResolveIssueUrl(expId), {
-          exp_issue_dict: issueToResolve.toBackendDict(),
-          exp_version: expVersion
-        }).then(function() {
-          var issueIndex = cachedIssues !== null ?
-            cachedIssues.findIndex(function(issue) {
-              return angular.equals(issue, issueToResolve);
-            }) : -1;
-          if (issueIndex === -1) {
-            var invalidIssueError = new Error(
-              'An issue which was not fetched from the backend has been ' +
-              'resolved');
-            return $q.reject(invalidIssueError);
-          } else {
-            cachedIssues.splice(issueIndex, 1);
-          }
-        });
-      },
-    };
-  }]);
+  private getFullIssuesUrl(explorationId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_ISSUES_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  private getFullPlaythroughUrl(expId: string, playthroughId: string): string {
```

Comment:
Please change `expId` to `explorationId`, we should keep the name consistent with all other usages.